### PR TITLE
Add depend_on_what_you_use 0.4.0

### PR DESCRIPTION
**Pull request by martis42**

### CODE REVIEW

File: modules/depend_on_what_you_use/0.4.0/MODULE.bazel [hunk 1]
```diff
@@ -0,0 +1,46 @@
+module(
+    name = "depend_on_what_you_use",
+    version = "0.4.0",
+    # Keep in sync with setup_step_2.bzl
+    bazel_compatibility = [">=5.4.0"],
+)
+
+bazel_dep(name = "rules_cc", version = "0.0.8")
```
Comment (Vertexwahn) [inline modules/depend_on_what_you_use/0.4.0/MODULE.bazel:8]: 0.0.9 is available already

File: modules/depend_on_what_you_use/0.4.0/MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+non_module_dependencies = use_extension("//third_party:extensions.bzl", "non_module_dependencies")
+use_repo(non_module_dependencies, "dwyu_pcpp")
+
+###
+### Development Dependencies
+###
+
+# Keep in sync with third_party/dependencies.bzl
+bazel_dep(name = "bazel_skylib", version = "1.7.0", dev_dependency = True)
```
Comment (Vertexwahn) [inline modules/depend_on_what_you_use/0.4.0/MODULE.bazel:19]: 1.7.1 is the newest one

File: modules/depend_on_what_you_use/0.4.0/MODULE.bazel (hunk 1 above)
Comment (martis42) [reply modules/depend_on_what_you_use/0.4.0/MODULE.bazel:8]: BCR maintainers educated me that one shall use the minimum required version as dependency, not the latest greatest one: https://github.com/bazelbuild/bazel-central-registry/pull/1533#discussion_r1503273082

Furthermore, to my understanding this file should be a copy of the [DWYU MODULE.bzl file](https://github.com/martis42/depend_on_what_you_use/blob/0.4.0/MODULE.bazel#L7) with patching for the version applied. DWYU 0.4.0 uses rules_cc 0.0.8.

File: modules/depend_on_what_you_use/0.4.0/MODULE.bazel (hunk 2 above)
Comment (martis42) [reply modules/depend_on_what_you_use/0.4.0/MODULE.bazel:19]: DWYU updates its dev dependencies just once a month to reduce noise by dependabot PRs.

Furthermore, to my understanding this file should be a copy of the [DWYU MODULE.bzl file](https://github.com/martis42/depend_on_what_you_use/blob/0.4.0/MODULE.bazel#L7) with patching for the version applied. DWYU 0.4.0 uses bazel_skylib 1.7.0.

**Review by fmeum — APPROVED** [review]

**Pull request merged**